bgpd/bgp_mplsvpn.c

```diff
@@ -638,7 +640,6 @@ void ensure_vrf_tovpn_sid(struct bgp *bgp_vpn, struct bgp *bgp_vrf, afi_t afi)
 	if (tovpn_sid_transpose_label == 0) {
 		zlog_debug("%s: not allocated new sid for vrf %s: afi %s",
 			   __func__, bgp_vrf->name_pretty, afi2str(afi));
-		XFREE(MTYPE_BGP_SRV6_SID, tovpn_sid_locator);
```

Comment:
Don't we need to free tovpn_sid_locator before returning?

Reply:
Yes exactly, I fixed.